shared/src/test/diff/pretyper/ucs/patterns/Literals.mls

```diff
@@ -10,25 +10,91 @@ class Pair[A, B](x: A, y: B)
 //│ class Pair[A, B](x: A, y: B)
 
 fun f(x) = if x is Some(1) then true else false
-//│ fun f: (Object & ~#Some | Some[Eql[1]]) -> Bool
+//│ fun f: (Object & ~#Some | Some[Object]) -> Bool
+
+[f(Some(1)), f(None), f(Some(2)), f(Some(-1))]
+//│ [Bool, Bool, Bool, Bool]
+//│ res
+//│     = [ true, false, false, false ]
+
+fun f(x) = if x is Some(1) then true
+//│ fun f: Some[1] -> true
+
+fun f(x) = if x is
+  Some(1) then true
+  Some(2) then false
+//│ fun f: Some[1 | 2] -> Bool
+
+[f(Some(1)), f(Some(2))]
+//│ [Bool, Bool]
+//│ res
+//│     = [ true, false ]
 
 fun g(x) = if x then 1 else 2
 //│ fun g: Bool -> (1 | 2)
 
 :e
 fun test_must_be_boolean(x) = if 0 then 1 else 2
 //│ ╔══[ERROR] Type mismatch in type ascription:
-//│ ║  l.19: 	fun test_must_be_boolean(x) = if 0 then 1 else 2
+//│ ║  l.37: 	fun test_must_be_boolean(x) = if 0 then 1 else 2
 //│ ║        	                                 ^
 //│ ╙── integer literal of type `0` is not an instance of type `Bool`
 //│ fun test_must_be_boolean: anything -> (1 | 2)
 
 fun g(x) = if x is true then 1 else 2
-//│ fun g: Eql[true] -> (1 | 2)
+//│ fun g: Object -> (1 | 2)
+
+[g(true), g(false), g(None)]
+//│ [1 | 2, 1 | 2, 1 | 2]
+//│ res
+//│     = [ 1, 2, 2 ]
 
 fun g(x) = if x && true is true then 1 else 2
 //│ fun g: Bool -> (1 | 2)
 
 fun h(x) = if (x : Bool) then 1 else 2
 //│ fun h: Bool -> (1 | 2)
 
+// Currently post-processing cannot hanlde this case. The desugared term is not
+// perfect. Also, is the inferred type wrong?
+fun mix(x) = if x is
+  true then "true"
+  Some(value) then "Some"
+  0 then "zero"
+//│ fun mix: (0 | Some[anything]) -> ("Some" | "true" | "zero")
+
+[mix(true), mix(Some(1)), mix(0)]
```

Comment:
This shouldn't type check according to the inferred signature above. Maybe a type simplification bug 🤔

Reply:
I posted the desugared term at the previous comment. I think it's a type simplification bug.